### DISCUSSION TITLE
AuthorizerConfigurationException from getOrders (Vendor Direct Fulfillment Orders API)

### DISCUSSION THREAD

**Discussion by virtutemdave** [saleweaver/python-amazon-sp-api#514] [Q&A]
I'm calling the Vendor Direct Fulfillment Orders API getOrders endpoint (/vendor/directFulfillment/orders/v1/purchaseOrders). This requires a Restricted Data Token (RDT). I am able to get the RDT, but when I call getOrders I get an AuthorizerConfigurationException. When I was debugging, I noticed that the URL created added multiple marketplaceId parameters. They all had the same value, ATVPDKIKX0DER for the US. Do you think that would cause this issue?

This is the generated URL:

```
/vendor/directFulfillment/orders/v1/purchaseOrders?createdAfter=2022-04-14T02%3A00%3A00Z&createdBefore=2022-04-14T02%3A59%3A59Z&MarketplaceId=ATVPDKIKX0DER&MarketplaceIds=ATVPDKIKX0DER&marketplace_ids=ATVPDKIKX0DER&marketplaceIds=ATVPDKIKX0DER
```

TIA,
David

**Comment by saleweaver** [comment]
I don't think that's causing the issue, it works fine for all other endpoints and is intended (accounts for all possible parameter name-styles/casing, Amazon ignores the others). 

Can you share how you're creating the rdt? Any errors there?

**Comment by virtutemdave** [comment]
I'm calling the Tokens endpoint to get the RDT. Here's the code for that. I'm sending the real LWA Client Id in my code.

```
restricted_token = Tokens(credentials=client_config).create_restricted_data_token(restrictedResources={
        "method": "GET",
        "path": "/vendor/directFulfillment/orders/v1/purchaseOrders",
        "dataElements": ["buyerInfo", "shippingAddress"]
    }, targetApplication='LWA Client Id')
```

I do not get any errors from this call and do get an RDT.

Thanks,
David
